packages/react/utils/src/createContext.tsx

```diff
@@ -12,3 +12,24 @@ export function createContext<ContextValueType>(displayName: string, rootCompone
   }
   return [Context, useContext] as const;
 }
+
+export function createContextObj<ContextValueType extends object>(rootComponentName: string) {
+  const Context = React.createContext<ContextValueType>(null as any);
+
+  function Provider(props: ContextValueType & { children: React.ReactNode }) {
+    const { children, ...providerProps } = props;
+    const value = React.useMemo(() => providerProps, [providerProps]) as ContextValueType;
```

Comment:
wait, this looks like it would re-memoize every render lol.. 🙈  hmm, give me a sec